src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -164,26 +164,26 @@ bool Pane::ResizePane(const Direction& direction)
     {
         return _Resize(direction);
     }
-    else
+
+    // If neither of our children were the focused leaf, then recurse into
+    // our children and see if they can handle the resize.
+    // For each child, if it has a focused descendant, try having that child
+    // handle the resize.
+    // If the child wasn't able to handle the resize, it's possible that
+    // there were no descendants with a separator the correct direction. If
+    // our separator _is_ the correct direction, then we should be the pane
+    // to resize. Otherwise, just return false, as we couldn't handle it
+    // either.
+    if ((!_firstChild->_IsLeaf()) && _firstChild->_HasFocusedChild())

```

Comment:
This pattern ("find the non-leaf's child, if it has one") appears a lot. Is there a way to pull it out into a helper? I know that we do slightly different things with each one, but the pattern looks the same:

```
if(!<x>.leaf && <x>.child) {
    make <x> do <A>, or do <_A> ourselves
}
```

perhaps we can:

```
auto x = _findFirstChildPaneMatchingThoseCriteriaOrNot();
return x.has_value() && x->A() || _A();
```

```